### ISSUE TITLE
Scrolling is laggy with high speed gaming mice

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#4997]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When users use Fyne apps with high-speed gaming mice, e.g. 1000 Hz, the scrolling is very laggy and continues to process a backlog of scroll events after the user stops dragging the scroll bar.

A possible solution is to merge scroll events in the driver if they arrive too fast so they are added together and processed only once per frame with the combined delta of all the scroll events that were merged.

### How to reproduce

Run any Fyne app with a scroll container using a mouse that reports very fast, say 1000 Hz

### Example code

fyne_demo

### Fyne version

develop

### Go compiler version

n/a

### Operating system and version

all

**Comment by dweymouth** [comment]
See [githu](https://github.com/dweymouth/supersonic/issues/418) for more details and a user-submitted screen capture

**Comment by andydotxyz** [comment]
I don't believe this can apply to all operating systems - the way they handle stepping is very different.
I'm guessing this is a Linux report - and may relate specifically to the X / Wayland / WM configuration.

Please include at least the minimal info for it to be reproducible.

In addition this seems to confuse two items - the refresh rate and the scroll speed. Scrolling is normally communicated through button presses, but the refresh rate of the mouse refers to the precision of the location on screen and speed of the move...

**Comment by dweymouth** [comment]
From the user reporting the bug to Supersonic:

OS: Arch Linux
DE: GNOME 46.3.1
WM: Mutter (Wayland)

They are also using xwayland as far as I can tell since I don't think a Wayland build is packaged in Arch

**Comment by Bush-cat** [comment]
> I don't believe this can apply to all operating systems - the way they handle stepping and scrolling is very different. I'm guessing this is a Linux report - and may relate specifically to the X / Wayland / WM configuration.

I'm having the same issue on windows, scrolling takes ages.

It's not as laggy as in the video below but laggy in a sense that it feels like it only runs at 30fps on my 144hz display, scrolling in any other app is much more responsive.

https://github.com/dweymouth/supersonic/assets/126769537/0e4ad93e-bfbc-4760-8715-35883f958412

_Originally posted by @kimjongbing in https://github.com/dweymouth/supersonic/issues/418#issuecomment-2222591874_

**Comment by dweymouth** [comment]
If you can, please chime in on the linked upstream issue in Fyne with your OS and more details, since the fix will have to happen there- thanks!

**Comment by andydotxyz** [comment]
I guess it would be helpful to get specifics on hardware or replication setup. On my high polling device the scroll events do not vary with the setting of the refresh rate.

**Comment by andydotxyz** [comment]
It would be worth checking this against develop, there are mentions from testers that the new thread model has helped a lot with scroll lag...

**Comment by redawl** [comment]
I think this is fixed by https://github.com/fyne-io/fyne/pull/5987, since it looks like from the user report that it was actually dragging the scroll bar with mouse movement that was causing the issue, not using the mouse scroll wheel to scroll.

**Comment by dweymouth** [comment]
Maybe, but the same optimization can probably be applied to scrolling as well. I'm not sure about the gaming mice, but Mac trackpads and the Apple mice definitely deliver scroll updates at more than 60 hz (I think 120 hz)
